### DISCUSSION TITLE
キーボードの片側が認識されない/TRRSケーブルの接続が不安定

### DISCUSSION THREAD

**Discussion by escape010** [Yowkees/keyball#313] [Q&A]
お世話になります。
質問を２つお願い致します。

1)キーボードの片側が認識されない

-片方をUSB-Cで繋ぎ、TRRSケーブルで左右を繋ぐ→USB-Cで繋いだ側のキーしか認識されない
-左をUSB-Cで繋ぐ（左右はTRRSケーブルで繋いだ状態）→左側のキーしか入力できず、右側は反応しない。
-右をUSB-Cで繋いだ場合は上記の逆となる。

左右どちらも全てのキーの動作確認はできている状態です。
ファームウェアは左右とも最終段階のものを書き込み済みです。

2)右側TRRSケーブルの接続が不安定
右側にTRRSケーブルを差し込み後、TRRSケーブルを少しでも動かすと右OLEDの表示が消えると同時に、
OLED裏の電通を示す赤いランプも消えてしまいます（写真２枚目から３枚目の状態になります）
→その後TRRSケーブルをはずすとOLEDが点灯する。

過去の質問などを参考に基盤裏のDATA部と、TRRSソケット部のハンダ付けを何度かやり直したのですが、
改善されないため、お手数ですが、対処方法を教えていただけますと幸いです。
![IMG_7911](https://github.com/Yowkees/keyball/assets/141485819/43ae41dc-bad0-4fdd-b341-bf0566e024ea)
![IMG_7914](https://github.com/Yowkees/keyball/assets/141485819/4caae6ef-ff6f-4b9a-9cfc-3d860e11af24)
![IMG_7915](https://github.com/Yowkees/keyball/assets/141485819/13f483b1-b3f2-4b0f-aa0d-237ff1e35c51)
![IMG_7916](https://github.com/Yowkees/keyball/assets/141485819/571f7db9-4e27-4df2-9b74-7630690bd70f)
![IMG_7917](https://github.com/Yowkees/keyball/assets/141485819/9c70fead-2c9b-4680-aa1e-436d8644efca)

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

本症状の多くは、DATA端子の半田不良もしくはDATA端子が近くの端子とショートしている場合に発生します。
すでに半田付けを見直していただけたようですが、【左右のProMicro裏のDATA端子どうし】が導通しているかご確認お願いします。

また、もしかするとTRSケーブルに問題がある可能性もあります。別のTRSケーブルに変えてみて症状が改善しないかもご確認いただければ幸いです。

以上、お手数ですがご確認お願いいたします。

**Reply by escape010** [reply]
ご回答ありがとうございます。

回答いただいた以下の内容について、です。
＞＞【左右のProMicro裏のDATA端子どうし】が導通しているかご確認お願いします。

こういった工作自体が初めてのため、
どの様に導通を確認すれば良いか（どの様な状態なら導通している、と判断できるのか）が理解できておりません。

導通できていれば、TRSケーブルの不良という事になるかと思いますが、
どの様に導通を確認すれば良いか（どの様な状態なら導通している、と判断できるのか）を
お手数ですが、教えていただけると幸いです。

よろしくお願いいたします。

**Comment by Yowkees** [comment]
了解しました。
下記画像２枚にそれぞれオレンジ点で示した端子があります。この２点にテスターを当てて、ピーと音が出れば導通しています。導通していればTRSケーブルや半田付けには不良はありません。

また、オレンジ点とその隣の端子（青点）がショートしてないこともご確認ください。

半田付けのテンプラ（接触不良）や半田カスによる予想外のショートは目視での発見は難しいので、テスターをご使用いただければと思います。
お手数ですがご検討よろしくお願いいたします。
![KB61R](https://github.com/Yowkees/keyball/assets/84209772/891fc775-8de3-4820-8cab-21e89a1947d0)
![KB61L](https://github.com/Yowkees/keyball/assets/84209772/d4c5ec77-cc45-4193-8455-26183af76731)

**Reply by escape010** [reply]
TRSケーブルを交換した所、問題なく稼働するようになりました。ありがとうございました。

**Comment by Yowkees** [comment]
無事に動くようになってよかったです！
ぜひ末永くご愛用いただければ幸いです。
